### PR TITLE
Add guidelines for article type: tutorial

### PR DESCRIPTION
**Pull request by lornajane**
We're ready to add the fourth Diataxis content type to DevPortal, so we need to include it in our guidelines. This also adds a template for structuring a tutorial.

### CODE REVIEW

**Comment by TibsAtWork** [comment]
I don't have any specific comments, other than to say that it does make it clear what a tutorial *is*, by explaining in the template,  which I find very helpful.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -226,3 +229,41 @@ Title template: *About [subject]* (if this is a background information for a tas
...
+'''''''''''''''''
+
+Title template: *[Verb] a [noun] with [insert list of technologies here]* , e.g. *Fit an OpenSearch box to your Flask app* or *Build a temperature monitoring dashboard with InfluxDB and Grafana*
+
+::
+
+   Tutorial title 
+   ###############
+
+   Explain what the user will be building in this tutorial. Include the problem we are solving and the components or tools that will be used to solve it. Remember that the example "problem" can be silly as long as it is easy to understand. Random pet name generator, office tea rota, plant watering system ... you decide. Whatever it is, the user knows where they are going and what they will see when they get there.
```
[the remark this answers is not in the document]
Comment (lornajane) [reply CONTRIBUTING.rst:242]: I think it has to be something that is easy to understand, but not necessarily realistic

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -226,3 +229,41 @@ Title template: *About [subject]* (if this is a background information for a tas
...
+
+   If there are tools, accounts, licenses or particular dependencies that need to be installed before we start, let's get those out of the way first. If it requires Python 22 and the user doesn't live in the future, they know not to start spinning up all the other pieces of the puzzle. Don't forget to prompt the user to sign up for Aiven with the free trial, if appropriate.
+
+   First component
+   ---------------
+
+   Introduce the first piece of the puzzle. Tutorials need to be as logical as possible, which means making as few steps, or as few context changes as possible. For example, don't download a data file, create a database, install some code dependencies, import the data into the file, set some code up locally and then go back to the database for credentials. Instead, make the database, get and import the data, then set up the code with dependencies and connection information to build the app.
+
+   At each stage, if the user can DO something, that builds confidence and a sense of achievement. So if there's a "SHOW TABLES" command or something else to affirm that something is going well, have them do that, and explain how it fits the big picture, before you move on.
```
[the remark this answers is not in the document]
Comment (lornajane) [reply CONTRIBUTING.rst:256]: I'm not sure it is _always_ helpful but definitely something to consider. I added this to the extra resources section!

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -226,3 +229,41 @@ Title template: *About [subject]* (if this is a background information for a tas
...
+
+   Tutorial title 
+   ###############
+
+   Explain what the user will be building in this tutorial. Include the problem we are solving and the components or tools that will be used to solve it. Remember that the example "problem" can be silly as long as it is easy to understand. Random pet name generator, office tea rota, plant watering system ... you decide. Whatever it is, the user knows where they are going and what they will see when they get there.
+
+   Pre-requisites
+   --------------
+
+   If there are tools, accounts, licenses or particular dependencies that need to be installed before we start, let's get those out of the way first. If it requires Python 22 and the user doesn't live in the future, they know not to start spinning up all the other pieces of the puzzle. Don't forget to prompt the user to sign up for Aiven with the free trial, if appropriate.
```
[the remark this answers is not in the document]
Comment (lornajane) [reply CONTRIBUTING.rst:247]: This is a really good point! Added another paragraph about assuming who the user is and writing for that audience.

**Review by TibsAtWork — COMMENTED** [review]
Looking good to me. I would be happy to work to this spec. And I don't see any changes needed to the language.

If @anelook approves, I think this is ready?

**Review by TibsAtWork — APPROVED** [review]
Approved by myself since @anelook is on holiday.

**Pull request merged**